### PR TITLE
Lesson 5 Exceptions exercise, Comprehensions lab, and Mailroom 3

### PR DESCRIPTION
**Pull request by brianminsk**
Finished Lesson 5 Exceptions exercise, Comprehensions lab, and Mailroom 3

### CODE REVIEW

File: students/brian_minsk/lesson05/Assignments/mailroom3.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    except AssertionError:
+        print("Report creation failed")
+        return
+
+    # print header
+    print("{:<26}| {:<11}| {:<10}| {:<12}".format("Donor Name", "Total Given",
+                                                  "Num Gifts", "Average Gift"))
+    # print header line
+    for __ in range(0, 65):
+        print("-", end="")
```
Comment (natasha-aleksandrova) [inline students/brian_minsk/lesson05/Assignments/mailroom3.py]: you can just do print("-" * 65)

File: students/brian_minsk/lesson05/Assignments/mailroom3.py [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+def create_report_rows():
+    """ Create the rows (as described in create_report()) with the data from
+    the donor db, sorted by total donation amount (computed by the sort_key
+    function). Return a list with each row as a list item.
+    """
+    sorted_donor_keys = sorted(donor_db, key=sort_key, reverse=True)
+    report_rows = [create_single_report_row(donor_db[donor_key])
+                   for donor_key in sorted_donor_keys]
+    try:
+        assert len(report_rows) == len(donor_db)
```
Comment (natasha-aleksandrova) [inline students/brian_minsk/lesson05/Assignments/mailroom3.py]: assert isn't typically used in "production" code

File: students/brian_minsk/lesson05/Assignments/mailroom3.py [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+    except AssertionError:
+        raise AssertionError
```
Comment (natasha-aleksandrova) [inline students/brian_minsk/lesson05/Assignments/mailroom3.py]: why catch if you're just going to raise the exception anyway?

Comment (brianminsk) [reply students/brian_minsk/lesson05/Assignments/mailroom3.py]: I'm a little confused by this. Can you please elaborate a little?

Comment (brianminsk) [reply students/brian_minsk/lesson05/Assignments/mailroom3.py]: :ok_hand: Oh, I get it now. I don't handle the exception other than raising it so it is useless ... 
Thanks for the comment!  :cowboy_hat_face:

Comment (natasha-aleksandrova) [reply students/brian_minsk/lesson05/Assignments/mailroom3.py]: cool 👍

**Pull request merged**